### DISCUSSION TITLE
Gradle AST Support

### DISCUSSION THREAD

**Discussion by shresthaujjwal** [openrewrite/rewrite#618] [Q&A] [answered]
Any plans to provide support for gradle ?

**Comment by jbduncan** [comment]
There is actually Gradle support already!

https://github.com/openrewrite/rewrite-gradle-plugin

I hope this helps. :)

**Reply by jkschneider** [reply]
As @jbduncan helpfully pointed out, the Gradle plugin helps to apply Rewrite recipes via a Gradle build task. Separately, there is a plan to support a Gradle AST (similar to the Maven AST) that allows you to refactor Gradle build scripts themselves.

As a historical note, my first foray into this kind of thing was with Nebula's [gradle-lint-plugin](https://github.com/nebula-plugins/gradle-lint-plugin), which to a limited extent may do what you are asking for. We've learned a lot since then. When Gradle support lands in OpenRewrite, it will be on a type-attributed Groovy AST that can be more accurate than the original work we did back then.

@shresthaujjwal It would be interesting to know whether Groovy or Kotlin build scripts are most prevalent for you, since we will have to develop ASTs for each.

**Reply by shresthaujjwal** [reply]
i should clarify, i was actually looking for `Gradle AST` to update Gradle Managed artifact version  and plugins etc just like maven pom.

**Reply by shresthaujjwal** [reply]
@jkschneider i am actually interested in Groovy build Scripts.

**Reply by jkschneider** [reply]
@shresthaujjwal Not _too_ far away from having this support now. The really exciting thing is that the Groovy AST (and therefore Gradle AST) will be able to run all the Java base recipes. Stay tuned!

**Reply by shresthaujjwal** [reply]
Hi just following up to see if there has been any progress on this

**Reply by timtebeek** [reply]
@shresthaujjwal please have a look at [the rewrite-gradle module](https://github.com/openrewrite/rewrite/tree/v7.35.0/rewrite-gradle/src/main/java/org/openrewrite/gradle) for the latest developments; there's been quite some progress over the past to years!

**Comment by timtebeek** [comment, marked as answer]
There's extensive support for Gradle nowadays in [the rewrite-gradle module](https://github.com/openrewrite/rewrite/tree/v7.35.0/rewrite-gradle/src/main/java/org/openrewrite/gradle), with [a collection of recipes as well](https://docs.openrewrite.org/reference/recipes/gradle).
